contracts/fee-manager/src/tests.rs

```diff
@@ -0,0 +1,157 @@
+                fee: Some(Decimal::percent(5)),
+                public_listing_fee: Some(Coin {
+                    denom: DENOM.to_string(),
+                    amount: Uint128::from(1000000u128),
+                }),
+            };
+
+            // Update.
+            app.execute_contract(
+                Addr::unchecked(ADMIN),
+                fee_manager_contract_addr.clone(),
+                &ExecuteMsg::Update {
+                    config: new_config.clone(),
+                },
+                &[],
+            )
+            .unwrap();
+
+            // Get config.
+            let response: ConfigResponse = app
+                .wrap()
+                .query_wasm_smart(fee_manager_contract_addr.clone(), &QueryMsg::GetConfig {})
+                .unwrap();
+            // Ensure fields updated.
+            assert_eq!(
+                response.config.receiver_addr.to_string(),
+                new_config.receiver_address.unwrap()
+            );
+            assert_eq!(response.config.fee, new_config.fee.unwrap());
+            assert_eq!(
+                response.config.public_listing_fee,
+                new_config.public_listing_fee.unwrap()
+            );
+        }
+
+        #[test]
+        fn updated_unauthorized() {
+            let (mut app, fee_manager_contract_addr) = proper_instantiate();
+
+            let new_config = ConfigUpdate {
+                receiver_address: Some(ADMIN.to_string()),
+                fee: Some(Decimal::percent(5)),
+                public_listing_fee: Some(Coin {
+                    denom: DENOM.to_string(),
+                    amount: Uint128::from(1000000u128),
+                }),
+            };
+
+            // Attempt update.
+            let err: ContractError = app
+                .execute_contract(
+                    // Send as USER instead of ADMIN.
+                    Addr::unchecked(USER),
+                    fee_manager_contract_addr.clone(),
+                    &ExecuteMsg::Update {
+                        config: new_config.clone(),
+                    },
+                    &[],
+                )
+                .unwrap_err()
+                .downcast()
+                .unwrap();
+            // Expect unauthorized.
+            assert_eq!(err, ContractError::Unauthorized {});
+
+            // Get config.
+            let response: ConfigResponse = app
+                .wrap()
+                .query_wasm_smart(fee_manager_contract_addr.clone(), &QueryMsg::GetConfig {})
+                .unwrap();
+            // Ensure fields not updated.
+            assert_ne!(
+                response.config.receiver_addr.to_string(),
+                new_config.receiver_address.unwrap()
+            );
+            assert_ne!(response.config.fee, new_config.fee.unwrap());
+            assert_ne!(
+                response.config.public_listing_fee,
+                new_config.public_listing_fee.unwrap()
+            );
```

Comment:
Stronger assertions here are likely that the fields equal the expected values, not that they do not equal the erroneously provided ones. If we assert that we can verify both that the new config was not applied and that existing data was not corrupted.